client/client-multi/client-cpp/test/impl/log/SQLiteDBLogStorageTest.cpp

```diff
@@ -108,51 +108,76 @@ BOOST_AUTO_TEST_CASE(GetAllRecordsTest)
 
     KaaUserLogRecord record;
     record.logdata = testLogData;
-    LogRecordPtr serializedRecord(new LogRecord(record));
+    LogRecord serializedRecord(record);
 
     std::size_t recordCount = 5;
     for (std::size_t i = 0; i < recordCount; ++i) {
-        logStorage.addLogRecord(serializedRecord);
+        logStorage.addLogRecord(std::move(serializedRecord));
     }
 
-    auto logs = logStorage.getRecordBlock(recordCount * serializedRecord->getSize(), mockBlocksCount);
+    auto logs = logStorage.getNextBucket();
 
-    BOOST_CHECK_EQUAL(logs.second.size(), recordCount);
+    BOOST_CHECK_EQUAL(logs.getRecords().size(), recordCount);
 
     std::size_t totalSize = 0;
-    for (auto& record : logs.second) {
-        totalSize += record->getSize();
+    for (auto& record : logs.getRecords()) {
+        totalSize += record.getSize();
     }
 
-    BOOST_CHECK_EQUAL(totalSize, recordCount * serializedRecord->getSize());
+    BOOST_CHECK_EQUAL(totalSize, recordCount * serializedRecord.getSize());
```

Comment:
Object was moved previously, but here it is used again without preceding assignment.